### ISSUE TITLE
Question: using snorkel for tabular mixed numeric and textual data to decide the label

### ISSUE THREAD

**Issue by Gooongna** [HazyResearch/snorkel#1056]
I have a task to label a large number of tabular records which is mixture with both numeric and text data and I already have some rule functions to decide the labels of each records. The generated labels could be conflict or correlated. Currently I am considering to use the data programming framework to generate the final probabilistic label.
1) Can I fit the Generative Model with the label matrix generated by my own rule functions outside of Snorkel?
2) Most of my rule functions decide the label based on the additional information from an external resource (table format). Not simply check whether the unlabeled data in external source but do some further operation based on additional information. In this case, is the data programming framework still workable? will the accuracy be affected because the data programming induce label based on statistic theory without raw data?

**Comment by paroma** [comment]
This is great question! 
1. Yes, you can run the Generative Model with your own label matrix!
2. The Generative Model doesn't not require the raw data and uses the label matrix directly to learn the accuracies and correlations of the labeling functions (more details [here](http://www.vldb.org/pvldb/vol11/p269-ratner.pdf))

**Comment by Gooongna** [comment]
Thanks for the quick response :)
On more further question: let assume the unlabeled training data is X, the number of training data with ground truth label is Y. If I learn and tuning the Generative Model by fitting with data Y, and apply the trained Generative model on data X. How large the data amount Y should  be, or what ratio X:Y should be? Is there any suggestion? Thanks.

**Comment by Michael870** [comment]
I have the same problem as @Gooongna that I need to label a large number of tabular records which is mixture with both numeric and text data. I learned that Candidate is an essential class in Snorkel API. How can I convert my tabular records in a CSV file to candidates?

**Comment by Michael870** [comment]
I have the same question that I need to apply Snorkel tabular mixed numerical and text data. Is it a feasible approach to rewrite the candidate_subclass() method to: 1) add attributes to the Candidate class for my numerical values; 2)add above attributes to DB table? Will this break the pipeline of defining LFs and generative model training?
